openupgradelib/openupgrade_merge_records.py

```diff
@@ -54,6 +58,9 @@ def _change_foreign_key_refs(env, model_name, record_ids, target_record_id,
             env.cr.execute('ROLLBACK TO SAVEPOINT sp1')
             if error.pgcode != UNIQUE_VIOLATION:
                 raise
+            elif error.pgcode == UNDEFINED_COLUMN:
```

Comment:
Isn't the exception not raised already at this point?
Also, I would prefer that you check if `extra_where` is actually set before you assume it's causing the error here.

Reply:
I don't get your first sentence, but yes, we can check if `extra_where` is set. Does this looks good to you:
```suggestion
            elif error.pgcode == UNDEFINED_COLUMN:
                if not extra_where:
                    raise
```

Reply:
Yes, agree with that.

What I mean with my first sentence, is that if `error.pgcode == UNDEFINED_COLUMN` is true, then `error.pgcode != UNIQUE_VIOLATION` is also true in which case line 60 raises before your code can take effect.

Reply:
I get you. What about now?

Reply:
Good now, thanks!